### PR TITLE
Code Refactoring: Move delta-sharing-client out as a separate directory

### PR DESCRIPTION
**Pull request by linzhou-db**
Code Refactoring: Move delta-sharing-client out as a separate directory

### CODE REVIEW

**Review by charlenelyu-db — APPROVED** [review]

File: client/src/main/scala/io/delta/sharing/client/filters/JsonPredicates.scala [hunk 1]
```diff
@@ -14,7 +14,7 @@
  * limitations under the License.
  */
 
-package io.delta.sharing.spark.filters
+package io.delta.sharing.client.filters
```
Comment (chakankardb) [inline client/src/main/scala/io/delta/sharing/client/filters/JsonPredicates.scala:17]: Why do we need to move the predicates?
The client treats them as a json string right?

File: client/src/main/scala/io/delta/sharing/client/filters/OpConverter.scala [hunk 2]
```diff
@@ -14,7 +14,7 @@
  * limitations under the License.
  */
 
-package io.delta.sharing.spark.filters
+package io.delta.sharing.client.filters
```
Comment (chakankardb) [inline client/src/main/scala/io/delta/sharing/client/filters/OpConverter.scala:17]: as above.

File: client/src/main/scala/io/delta/sharing/client/util/ConfUtils.scala [hunk 3]
```diff
@@ -14,7 +14,7 @@
  * limitations under the License.
  */
 
-package io.delta.sharing.spark.util
+package io.delta.sharing.client.util
```
Comment (chakankardb) [inline client/src/main/scala/io/delta/sharing/client/util/ConfUtils.scala:17]: This seems odd. These are all spark confs. Why should they belong int the client?

Comment (linzhou-db) [reply client/src/main/scala/io/delta/sharing/client/util/ConfUtils.scala:17]: ConfUtils  is actually used by DeltaSharingFileSystem, and used to initialize DeltaSharingRestClient in RemoteDeltaLog, I'll refactor that later as well.

File: client/src/main/scala/io/delta/sharing/client/filters/JsonPredicates.scala (hunk 1 above)
Comment (linzhou-db) [reply client/src/main/scala/io/delta/sharing/client/filters/JsonPredicates.scala]: good point. moved back to spark.

File: client/src/main/scala/io/delta/sharing/client/filters/OpConverter.scala (hunk 2 above)
Comment (linzhou-db) [reply client/src/main/scala/io/delta/sharing/client/filters/OpConverter.scala]: good point. moved back to spark.

File: client/src/main/scala/io/delta/sharing/client/util/ConfUtils.scala (hunk 3 above)
Comment (chakankardb) [reply client/src/main/scala/io/delta/sharing/client/util/ConfUtils.scala:17]: Having DeltaSharingFileSystem in the client seems a bit odd.
Why do we need it in the client?

Comment (linzhou-db) [reply client/src/main/scala/io/delta/sharing/client/util/ConfUtils.scala:17]: It was discussed in the design doc actually. 
delta-sharing-client serves two main purposes: 1) DeltaSharingClient to handle http requests/responses 2) DeltaSharingFileSystem to handle reading data out of pre-signed urls.

**Review by chakankardb — APPROVED** [review]

**Pull request merged**